packages/engine/Source/Scene/ArcGisMapServerImageryProvider.js

```diff
@@ -398,6 +422,87 @@ function ArcGisMapServerImageryProvider(options) {
   }
 }
 
+/**
+ * Gets the credits to be displayed when a given tile is displayed.
+ * @param {object} [options] Object with the following properties:
+ * @param {ArcGisBaseMapType} [options.style=ArcGisBaseMapType.SATELLITE] The style of the ArcGIS base map imagery. Valid options are {@link ArcGisBaseMapType.SATELLITE}, {@link ArcGisBaseMapType.OCEANS} and {@link ArcGisBaseMapType.HILLSHADE}.
+ * @returns {ArcGisMapServerImageryProvider}
+ */
+
+ArcGisMapServerImageryProvider.fromBasemapType = function (options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  let accessToken;
+  let server;
+  let warningCredit;
+  switch (options.style) {
+    case ArcGisBaseMapType.SATELLITE:
+      {
+        accessToken = defaultValue(
+          options.token,
+          ArcGISMapServiceWorldImagery.defaultAccessToken
+        );
+        server = Resource.createIfNeeded(
+          defaultValue(options.url, ArcGISMapServiceWorldImagery.defaultServer)
+        );
+        server.appendForwardSlash();
+        const defaultTokenCredit = ArcGISMapServiceWorldImagery.getDefaultTokenCredit(
+          accessToken
+        );
+        if (defined(defaultTokenCredit)) {
+          warningCredit = Credit.clone(defaultTokenCredit);
+        }
+      }
+      break;
+    case ArcGisBaseMapType.OCEANS:
+      {
+        accessToken = defaultValue(
+          options.token,
+          ArcGISMapServiceWorldOcean.defaultAccessToken
+        );
+        server = Resource.createIfNeeded(
+          defaultValue(options.url, ArcGISMapServiceWorldOcean.defaultServer)
+        );
+        server.appendForwardSlash();
+        const defaultTokenCredit = ArcGISMapServiceWorldOcean.getDefaultTokenCredit(
+          accessToken
+        );
+        if (defined(defaultTokenCredit)) {
+          warningCredit = Credit.clone(defaultTokenCredit);
+        }
+      }
+      break;
+    case ArcGisBaseMapType.HILLSHADE:
+      {
+        accessToken = defaultValue(
+          options.token,
+          ArcGISMapServiceWorldHillshade.defaultAccessToken
+        );
+        server = Resource.createIfNeeded(
+          defaultValue(
+            options.url,
+            ArcGISMapServiceWorldHillshade.defaultServer
+          )
+        );
+        server.appendForwardSlash();
+        const defaultTokenCredit = ArcGISMapServiceWorldHillshade.getDefaultTokenCredit(
+          accessToken
+        );
+        if (defined(defaultTokenCredit)) {
+          warningCredit = Credit.clone(defaultTokenCredit);
+        }
+      }
+      break;
+    default:
+  }
+
+  return new ArcGisMapServerImageryProvider({
+    options,
```

Comment:
I think you intended to use the spread operator in order to pass the provided options other than the overridden ones below.

```suggestion
    ...options,
```